toucan_connectors/one_drive/one_drive_connector.py

```diff
@@ -192,10 +222,13 @@ def _retrieve_data(self, data_source: OneDriveDataSource) -> pd.DataFrame:
             cols = data[0]
             data.pop(0)
 
-            df_sheet_current = pd.DataFrame(data, columns=cols)
-            if len(sheet_list) > 1:
-                df_sheet_current['__sheetname__'] = sheet
+            df_current = pd.DataFrame(data, columns=cols)
+            if len(workbook_elements_list) > 1:
+                if data_source.sheet:
+                    df_current['__sheetname__'] = workbook_element
+                else:
+                    df_current['__tablename__'] = workbook_element
 
```

Comment:
So that here, you're going to have : 
```
df_current[workbook_key_column] = workbook_element
```
instead of checking again if/else on `data_source.sheet` or `data_source.table`